utils/socket.c

```diff
@@ -127,3 +127,28 @@ int agent_message_read_head(int fd, struct uftrace_msg *msg)
 
 	return 0;
 }
+
+/**
+ * agent_message_read_response - read agent ack
+ * @fd - socket file descriptor
+ * @response - ack from agent
+ * @return - status: data or error code, negative on error
+ */
+int agent_message_read_response(int fd, struct uftrace_msg *response)
+{
+	int status = 0;
+
+	if (agent_message_read_head(fd, response) < 0)
+		return -1;
+
+	if (response->len > sizeof(status))
```

Comment:
This is just the current implementation issue.  I think we can have a response with bigger data payload.

Reply:
I wanted to keep things simple and only return an `int` in response messages. Should I allocate memory here for arbitrary data?

Reply:
No, we can do that later if needed.